pkg/provider/jenkins/README.md

```diff
@@ -0,0 +1,93 @@
+configured Jenkins job(s). The server will attempt to send an HTTP request to each defined Jenkins job, and it will
+report each job as "healthy" if the request is successful and the job health status matches one of the expected values,
+or "unhealthy" otherwise.
+
+## Configuration
+
+The Jenkins Provider is configured through the platform-health server's configuration file, with component instances
+listed under the `jenkins` key.
+
+* `job` (optional): The name of the Jenkins job, used to identify the service in the health reports.
+* `url` (required): The URL of the Jenkins controller service to monitor.
+* `timeout` (default: `10s`): The maximum time to wait for a response before timing out.
+* `insecure` (default: `false`): If set to true, allows the Jenkins provider to establish connections even if the HTTP
+  certificate of the service is invalid or untrusted. This is useful for testing or in environments where services use
+  self-signed certificates. Note that using this option in a production environment is not recommended, as it disables
+  important security checks.
+* `status` (default: `["Succeeded"]`): The list of desired job statuses that are expected in the response.
+* `detail` (default: `false`): If set to true, the provider will return detailed information about the Jenkins job.
+* `credentials` (optional): The credentials required to authenticate with the Jenkins controller. This is an array of
+  credential objects which aims at populating the `username` and `token` fields. Multiple credentials providers can be used.
+  If multiple credentials providers are used, the `username` and `token` fields are populated by merging the provider results.
+
+## Credentials provider support
+
+|             Supported             | Provider      | Description                                                                                 |
+|:---------------------------------:|---------------|---------------------------------------------------------------------------------------------|
+|               @TODO               | `environment` | Reads the credentials from environment variables.                                           |
+|               @TODO               | `vault`       | Reads the credentials from HashiCorp Vault. Currently supported engines are `kv` v1 and v2. |
+
+### Example
+
+```yaml
+jenkins:
+  - job: example
+    url: https://jenkins.com
+    detail: true
+```
+
+### Example with credentials (environment variables)
+
+```yaml
+jenkins:
+  - job: example
+    url: https://jenkins.com
+    detail: true
+    credentials:
+      - provider: environment
+        username: JENKINS_USERNAME
+        token: JENKINS_TOKEN
+```
+
+### Example with credentials (HashiCorp Vault)
+
+```yaml
+jenkins:
+  - job: example
+    url: https://jenkins.com
+    detail: true
+    credentials:
+      - provider: vault
+        address: https://vault.com
```

Comment:
Were we to add a Vault credentials source, then I think that address would belong in it, rather than per (component-)provider-instance =)